pkg/plugins/common/kustomize/v2/scaffolds/internal/templates/config/prometheus/monitor_tls_patch.go

```diff
@@ -44,24 +44,21 @@ func (f *ServiceMonitorPatch) SetTemplateDefaults() error {
 
 const serviceMonitorPatchTemplate = `# Patch for Prometheus ServiceMonitor to enable secure TLS configuration
 # using certificates managed by cert-manager
-apiVersion: monitoring.coreos.com/v1
-kind: ServiceMonitor
-metadata:
-  name: controller-manager-metrics-monitor
-  namespace: system
-spec:
-  endpoints:
-    - tlsConfig:
-        insecureSkipVerify: false
-        ca:
-          secret:
-            name: metrics-server-cert
-            key: ca.crt
-        cert:
-          secret:
-            name: metrics-server-cert
-            key: tls.crt
-        keySecret:
-          name: metrics-server-cert
-          key: tls.key
+- op: replace
+  path: /spec/endpoints/0/tlsConfig
+  value:
+    # SERVICE_NAME and SERVICE_NAMESPACE will be substituted by kustomize
+    serverName: SERVICE_NAME.SERVICE_NAMESPACE.svc
```

Comment:
Did we only need to change the kustomize default file just to pass the serviceName right?
But that is fixed. It will always be the same. 
Could we not simplify the changes and the default kustomize by adding here the name

```suggestion
    serverName: controller-manager-metrics-monitor.SERVICE_NAMESPACE.svc
```

Instead?

Reply:
I am trying to see if we can avoid do changes at: [config/default/kustomization.yaml](https://github.com/kubernetes-sigs/kubebuilder/pull/4536/files#diff-82cfd4563abc54cb08f9a19275ffbb11c8ffbf1410304014e722c2685327300d)

Reply:
I tried to follow the existing convention followed in kubebuilder kustomize files.
For other scenarios where the service name is referred, we follow similar convention of replacements. See [certificate](https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugins/common/kustomize/v2/scaffolds/internal/templates/config/certmanager/certificate_metrics.go#L62)
I'm not sure if this `serverName` replacement is any different. Let me know if my understanding is wrong

Reply:
I am not too happy with the complexities in the kustomize but I think that is not part of the scope and its make sense. Thank you.